crates/runtime/src/traphandlers.rs

```diff
@@ -386,12 +386,29 @@ mod tls {
             })
         }
 
+        #[inline(never)]
+        /// Eagerly initialize thread-local runtime functionality. This will be performed
+        /// lazily by the runtime if users do not perform it eagerly.
+        pub fn initialize() -> Result<(), Trap> {
+            PTR.with(|p| {
+                let (state, mut initialized) = p.get();
+                if !initialized {
+                    super::super::sys::lazy_per_thread_init()?;
+                    initialized = true;
+                }
+                p.set((state, initialized));
+                Ok(())
```

Comment:
This could probably be a bit simpler with `if initialized { return }` and then `p.set((state, true))` afterwards